cfg/cppcheck-cfg.rng

```diff
@@ -352,14 +355,14 @@
               <zeroOrMore>
                 <element name="function">
                   <attribute name="name"><ref name="DATA-NAME"/></attribute>
-                  <choice>
-                    <attribute name="action">
-                      <ref name="CONTAINER-ACTION"/>
-                    </attribute>
+                  <attribute name="action">

```

Comment:
why are you making these changes here? with this change the "action" attribute is not optional anymore.

Reply:
The insert methods return iterators to the newly inserted elements for multimap or multiset, whereas for map or set they return a pair of a bool or iterator. I use this information to distinguish between multimap and map. Additionally, this will add more information about the methods in the AST.

So I need to use the `action` attribute as well as the `yields` attribute together. Is this the correct approach?

Reply:
These changes makes the `action` attribute required. Somehow we want to have one or both of these attributes.

Either make both attributes optional.. that is fine imho we don't need to be superstrict.

Or I guess this might work:

    <oneOrMore>
        <choice>
            ... old code ...
        </choice>
    </oneOrMore>

This relaxng means that we can have duplicate attributes but as far as I know that breaks the XML rules so we do not need to worry about that.

Reply:
> Either make both attributes optional.. that is fine imho we don't need to be superstrict.

hmm that is the approach you used ... that should work fine as far as I know.